wgengine/netstack/netstack.go

```diff
@@ -242,14 +242,28 @@ func (ns *Impl) acceptTCP(r *tcp.ForwarderRequest) {
 
 func (ns *Impl) forwardTCP(client *gonet.TCPConn, address string) {
 	ns.logf("[v2] netstack: forwarding to address %s", address)
-	server, err := ns.dialTCP(address)
+	ctx, cancel := context.WithCancel(context.Background())
+	server, err := ns.dialContextTCP(ctx, address)
 	if err != nil {
 		ns.logf("netstack: could not connect to server %s: %s", address, err)
 		client.Close()
+		cancel()
 		return
 	}
-	go io.Copy(server, client)
-	go io.Copy(client, server)
+	done := make(chan bool)
```

Comment:
this is an unbuffered channel and you send twice, which means one of those two goroutines will forever block and thus leak the goroutine (and its memory) forever. make it `make(chan bool, 2)` for clarity, even if 1 is probably sufficient.